src/pages/api/paymentTxParams/[uuid].ts

```diff
@@ -11,47 +13,25 @@ export default async function handler(req: NextApiRequest, res: NextApiResponse)
     try {
       const { uuid, buyerAddress } = req.query;
 
-      if (typeof uuid !== 'string') {
+      if (!uuid || typeof uuid !== 'string') {
         return res.status(400).json({ message: 'Invalid UUID' });
       }
 
+      // TODO: Justin / Rafa is this stil buyer? or sender? I dont remember
       if (!!buyerAddress && ((typeof buyerAddress !== 'string') || !ethers.utils.isAddress(buyerAddress))) {
         return res.status(400).json({ message: 'Invalid buyer address' });
       }
       
-      const paymentTxPromise = prisma.paymentTx.findUnique({
-        where: { uuid },
-      });
-      const txDataPromise = prisma.contactlessPaymentTxData.findUnique({
-        where: { uuid },
-      });
-      const txMessagePromise = prisma.contactlessPaymentMessage.findUnique({
+      const paymentTxOrMsg = await prisma.contactlessPaymentTxOrMsg.findUnique({
         where: { uuid },
       });
       
-      const [paymentTx, txData, txMessage] = (await Promise.allSettled([paymentTxPromise, txDataPromise, txMessagePromise])).map(promiseResult => {
-        if (promiseResult.status === 'fulfilled') {
-          return promiseResult.value
-        };
-
-        return undefined;
-      });
 
-      if (!txData && !paymentTx && !txMessage) {
+      if (!paymentTxOrMsg) {
         return res.status(404).json({ message: 'Not Found' });
       }
 
-      if (paymentTx) {
-        res.status(200).json({ payloadType: 'eip681', ...paymentTx });
-      } else if (txData) {
-        // Substitute the buyer address into the tx call data if applicable
-        const formattedTxData = await formatTxDataResponse({ txData, buyerAddress });
-        res.status(200).json({ payloadType: 'contractCall', ...formattedTxData })
-      } else {
-        // Substitute the buyer address into the message's from field if applicable
-        const formattedTxMessage = formatTxMessageResponse({ txMessage, buyerAddress });
-        res.status(200).json({ payloadType: 'eip712', ...formattedTxMessage });
-      }
+      res.status(200).json(paymentTxOrMsg);
```

Comment:
Can we spread the `txParams` field when we return the JSON object? I get that we need `txParams` to store the JSON fields, but I'd like to comply with what we have in the ERC spec right now:

This is what I get for an EIP-681 response with these changes:
```
{
    "uuid": "1759592d-73a6-4f9f-b334-160e9e1faca6",
    "verificationCode": "dGiul3Lv",
    "payloadType": "eip681",
    "chainId": "8453",
    "txParams": {
        "value": "10000",
        "toAddress": "0xf7d07Ee99095FDC09001710Ec232162a788BB989",
        "contractAddress": "0x833589fcd6edb6e08f4c7c32d4f71b54bda02913"
    },
    "rpcProxySubmissionParams": null,
    "dappName": null,
    "dappUrl": null,
    "createdAt": "2024-09-25T21:00:05.594Z",
    "updatedAt": "2024-09-25T21:00:05.595Z"
}
```

and this is what I'd be expecting:
```
{
    "uuid": "1759592d-73a6-4f9f-b334-160e9e1faca6",
    "verificationCode": "dGiul3Lv",
    "payloadType": "eip681",
    "chainId": "8453",
    "value": "10000",
    "toAddress": "0xf7d07Ee99095FDC09001710Ec232162a788BB989",
    "contractAddress": "0x833589fcd6edb6e08f4c7c32d4f71b54bda02913"
    "rpcProxySubmissionParams": null,
    "dappName": null,
    "dappUrl": null,
    "createdAt": "2024-09-25T21:00:05.594Z",
    "updatedAt": "2024-09-25T21:00:05.595Z"
}
```

Reply:
Sure thing! 🫡